packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx

```diff
@@ -99,7 +101,10 @@ export const ReferenceArrayInput = (props: ReferenceArrayInputProps) => {
     return (
         <ResourceContextProvider value={reference}>
             <ChoicesContextProvider value={controllerProps}>
-                {children}
+                {React.cloneElement(children, {
+                    disabled: children.props.disabled || disabled || readOnly,
+                    readOnly: children.props.readOnly || readOnly,
+                })}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** Not sure about that since [the React doc warns](https://react.dev/reference/react/cloneElement#alternatives) about using `cloneElement`‏. Can't we use another approach @djhi?